omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/PatientAllergyResource2_0.java

```diff
@@ -0,0 +1,246 @@
+			description.addProperty("allergen");
+			description.addProperty("severity");
+			description.addProperty("comment");
+			description.addProperty("reactions");
+			description.addProperty("patient");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			return description;
+		} else if (rep instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			// what should be in full representation
+			description.addProperty("display");
+			description.addProperty("uuid"); // always null, allergy does not support uuid
+			description.addProperty("allergyId");
+			description.addProperty("allergen");
+			description.addProperty("severity");
+			description.addProperty("comment");
+			description.addProperty("reactions");
+			description.addProperty("patient");
+			description.addProperty("voided");
+			description.addProperty("auditInfo");
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getCreatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		//description.addProperty("uuid"); // always null, allergy does not support uuid
+		description.addProperty("allergyId");
+		description.addProperty("allergen");
+		description.addProperty("severity");
+		description.addProperty("comment");
+		description.addProperty("reactions");
+        return description;
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getUpdatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		return getCreatableProperties();
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingSubResource#getParent(java.lang.Object)
+	 */
+	@Override
+	public Patient getParent(Allergy instance) {
+		return instance.getPatient();
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingSubResource#setParent(java.lang.Object,
+	 *      java.lang.Object)
+	 */
+	@Override
+	public void setParent(Allergy instance, Patient patient) {
+		instance.setPatient(patient);
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.SubResource#doGetAll(java.lang.Object,
+	 *      org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	public NeedsPaging<Allergy> doGetAll(Patient parent, RequestContext context) throws ResponseException {
+		List<Allergy> allergiesList = new ArrayList<Allergy>();
+		Allergies allergies = new Allergies();
+		if (parent != null) {
+			allergies = Context.getPatientService().getAllergies(parent);
+			if (allergies.getAllergyStatus().equals(Allergies.UNKNOWN)) {
+				// return 4040
```

Comment:
Make this comment more explicit. Say:

    // return a 404 status for a patient whose allergy status is unknown